src/d/semantic/caster.d

```diff
@@ -299,7 +299,7 @@ struct Caster(bool isExplicit, alias bailoutOverride = null) {
 					return CastKind.UPad;
 				}
 
-				return CastKind.Invalid;
+				goto case Byte;
```

Comment:
This does not seem related, nor does it look like it is covered by tests.

Reply:
dmd lets you cast bools to floats for some reason

Reply:
```d
if (isFloat(bt)) {
  return CastKind.SignedToFloat;
}
```

Maybe is it even worth adding a BoolToFloat cast type? SDC's internal generally don't consider bool to be an integral.

Reply:
Could do.

I went for the goto because the logic then flows the same way as the chars

Reply:
Clearly not, because you had to add an `if (t == Bool) {` in there.

Reply:
not anymore

Reply:
I didn't initially spot that `t` gets mutated

Reply:
It's far from a given that this is correct, as now the bailout will use ubyte.

Reply:
changed to an if statement